src/universal/utils/isObject.js

```diff
@@ -1,3 +1,3 @@
 export default function isObject(val) {
-  return val && typeof val === 'object';
+  return val && typeof val === 'object' && !(val instanceof Date);
```

Comment:
Ok, go for it. I'll hold off on merge until you refactor. I'm going to update `CHANGELOG`